### ISSUE TITLE
 Tuya TS0601 Radiator Valve - Unable to set temperature (worked before with customize)

### ISSUE THREAD

**Issue by Plawasan** [nielsfaber/scheduler-card#381]
I have been using the scheduler/card for controlling my Tuya TS0601 radiator valves using this customization for controlling their temperature set point:

```
   customize:
      climate.kitchen_radiator:
        actions:
          - service: climate.set_temperature
            name: Set temperature
            icon: fire
            variable:
              field: temperature
              name: Temperature
              min: 10
              max: 30
              step: 0.5
        exclude_actions:
          - heat
          - cool
```
This has been working until recently (can't really pinpoint until when exactly) however now I cannot set the temperature using the card plus there's a whole list of "Set Temperature" options that seemingly do nothing (there's no temp slider)

![image](https://user-images.githubusercontent.com/73796046/117043508-9cb1a500-ad0d-11eb-8bc4-66601abfe2ec.png)

When I tried to recreate a schedule I no longer have the option to use "Set Temperature" at all (despite the customization still in place), I can only use "Heat" mode with a set temperature however that results in the valve going into a force_open status at 100%. regardless of current vs. set temperature.

![image](https://user-images.githubusercontent.com/73796046/117043691-d97d9c00-ad0d-11eb-906e-fcb9297d7fbb.png)

I did try to remove the customization, no effect..

The problem seems to be limited to the card itself as the underlying schedules still run normal (which is why I can't really tell when did it go wrong). I am on the latest version of both the card (v2.1.1) and the scheduler (v3.1.0), I did try to rollback in steps all the way down to 2.0.3 without any difference in the behavior. Out of interest I went back to 3.0.5 for the scheduler, still no difference

Luckily have daily backups of configuration so I used those to restore my scheduler.storage and go back to the working config however I still cannot edit it in lovelace.

Just to confirm -  climate.set_temperature still works with the valve as expected when I tested it from developer tools.

Thanks for looking into this, please let me know if there's anything else you need me to provide.

**Comment by nielsfaber** [comment]
Change
```
            variable:
              field: temperature
              name: Temperature
              min: 10
              max: 30
              step: 0.5
```
to:
```
            variables:
             temperature:
                name: Temperature
                min: 10
                max: 30
                step: 0.5
```

still I don’t like how things show up for you now.
Could you share the configuration from scheduler.storage of this schedule which gives you the 6x “set temperature”  ?
I would like to use it as a test candidate.

**Comment by Plawasan** [comment]
Thanks.. I missed that when I was looking for open issues :/

However - it still didn't help. Current customize config:

```
    customize:
      climate.kitchen_radiator:
        actions:
          - service: climate.set_temperature
            name: Set temperature
            icon: fire
            variables:
              temperature:
                name: Temperature
                min: 10
                max: 30
                step: 0.5
        exclude_actions:
          - heat
          - cool
```
But I'm still not seeing the option to set temperature when creating a new schedule.

![image](https://user-images.githubusercontent.com/73796046/117047422-25cadb00-ad12-11eb-91e1-e4a7966c7dfa.png)

My scheduler.storage is here (I manually took out some unrelated schedules so hopefully formatting is still OK)

https://gist.github.com/Plawasan/b969e27f7d34143491e9ad2c15766489

**Comment by nielsfaber** [comment]
I managed to reproduce the issue and worked on some improvements.
It would be nice if you could test the improvements locally.
For doing so, please update your local scheduler-card installation to the attached file (see below).
It should be placed in `www/community/scheduler-card` in case you use HACS, otherwise you probably know the location yourself.
After overwriting the .js file, reload the frontend resources (click "reload resources" on the top right menu in HA lovelace).
Please check your [browser logs](https://zapier.com/help/troubleshoot/behavior/view-and-save-your-browser-console-logs), if the update went right it should show this badge:
![image](https://user-images.githubusercontent.com/16861605/117099323-247ccb00-ad71-11eb-9a6c-3c802c2c899a.png)

I noticed that your temperature variable misses a `unit: °C`, I suggest to add it.
It could be that this was auto-discovered by the previous version of the card, but I made quite some radical changes in the meantime which changed this behaviour.
I also would like to know why you need the customized action. I would assume that the scheduler offers an "heat" action out of the box that should work fine.

[scheduler-card.zip](https://github.com/nielsfaber/scheduler-card/files/6425136/scheduler-card.zip)

**Comment by Plawasan** [comment]
Thanks for the quick response

- I downloaded and manually updated the card however I can't get it to load. I'm running 2021.4.6 in docker and I simply don't see an option to reload resources from the UI. I even tried calling lovelace.reload_resources directly (per HA docs) but that service doesn't exist in my instance. I then deleted the .js, reloaded, made sure I get an error where the card should be and then copied the test one back in - still no Test version in the console. I double checked the file, it's the correct version:
```
pi@raspberrypi:~/homeassistant/www/community/scheduler-card $ cat scheduler-card.js | grep VERSION
    const CARD_VERSION = 'TEST';

```
- I deleted my browser cache, shift+f5, nothing seems to work
- When I check the URL of the scheduler.js file that reports the version to the console I get http://pi.lan:8123/hacsfiles/scheduler-card/scheduler-card.js however when I delete www/community/scheduler-card/scheduler-card.js that file goes 404 so I assume it's just a link to the same thing...

I'm baffled...

**Comment by nielsfaber** [comment]
I also run docker, that doesn't matter.
Just use the menu on the top right:
![image](https://user-images.githubusercontent.com/16861605/117109238-a4606080-ad84-11eb-97a3-16e70d86cad4.png)

Otherwise, restarting HA should also refresh the resources.

**Comment by Plawasan** [comment]
First thing I tried...

![image](https://user-images.githubusercontent.com/73796046/117109479-05883400-ad85-11eb-89f4-2c729c15123c.png)


![image](https://user-images.githubusercontent.com/73796046/117109452-f99c7200-ad84-11eb-9f55-dfc78130dd03.png)



![image](https://user-images.githubusercontent.com/73796046/117109426-edb0b000-ad84-11eb-82d3-5f7b48adef0d.png)

**Comment by Plawasan** [comment]
In the meantime to answer "_I also would like to know why you need the customized action. I would assume that the scheduler offers an "heat" action out of the box that should work fine_."

To my best knowledge it's related to this issue in zigbee2mqtt where using Heat mode forces the valve 100% open instead of setting the temperature - https://github.com/Koenkk/zigbee2mqtt/issues/5578 

So while that is still the case the customize: workaround is the only way to correctly set temperature for these valves.

**Comment by Plawasan** [comment]
OK.. don't ask me how but I managed to get to the test version:
![image](https://user-images.githubusercontent.com/73796046/117135646-87895480-ada7-11eb-8db9-bfa41ef9e2ba.png)

TEST **without customization** (what worked last in 2.0.5, stopped in 2.1.0) - no difference
![image](https://user-images.githubusercontent.com/73796046/117135748-a851aa00-ada7-11eb-99bb-d0624e1dc062.png)

TEST **with customization** - works OK (I did add the units however I don't see any impact of that)
```
      climate.bedroom_radiator:
        actions:
          - service: climate.set_temperature
            name: Set temperature
            icon: fire
            variables:
              temperature:
                name: Temperature
                min: 10
                max: 30
                step: 0.5
                unit: °C
```
![image](https://user-images.githubusercontent.com/73796046/117136799-28c4da80-ada9-11eb-8999-bf4fdb785223.png)

Remaining Issues with TEST:
- Schedule with multiple entities still doesn't work:
- ![image](https://user-images.githubusercontent.com/73796046/117136556-d1266f00-ada8-11eb-88d5-fdd25982e26e.png)

- Scheduler card doesn't show the mdi:temperature icon for schedules using Set Temperature (Bathroom Floor is a different thermostat using Heat)
- ![image](https://user-images.githubusercontent.com/73796046/117136663-f74c0f00-ada8-11eb-8d72-68cead275e1e.png)

**Comment by nielsfaber** [comment]
@Plawasan Thanks for the extensive reporting! 

> TEST without customization (what worked last in 2.0.5, stopped in 2.1.0) - no difference

I don't know what should be expected of scheduler-card in such scenario.
You create a customized task and later you remove it from the card config, how could an earlier created schedule still show up pretty?
It will never be able to give you a slider because the card doesn't know min/max values. It doesn't even know that temperature is variable (it considers it as a fixed parameter). Ideally I would see "set temperature to X", so you know that each button has a different temperature assigned to it, but this is not that easy to do.
I think this is a corner-case, I would say the user is reponsible for keeping customizations in place.

> TEST with customization - works OK (I did add the units however I don't see any impact of that)

Good! This is a big improvement compared to yesterday already. Without the units you wouldn't see the '°C' I guess, only "set to 19".

> Schedule with multiple entities (identical valves with identical customization) still doesn't work:

This looks like homework for me. I think that should be fixable.

> Scheduler card doesn't show the mdi:thermometer icon for schedules using Set Temperature (Bathroom Floor is a different thermostat using Heat)

You can simply assign an icon to the customized action.
(Again, scheduler cannot guess which icon you would like to see here)

> Scheduler card doesn't show the target temperature of the next {action}

As before, I would like to see "set temperature to X" here (these cases are related to eachother).
I would have to look into this, I think some things need to be restructured to make this come back.
But I agree with you, it doesn't look OK now.

I need a few days to shuffle things up to make things the way they were.
Sorry for breaking things.
I will keep you posted, hopefully you are willing to test again (and the updating goes more smoothly).

**Comment by Plawasan** [comment]
> @Plawasan Thanks for the extensive reporting!

No problem, I appreciate the work you're putting into this, the least I can do is try to help.

> I don't know what should be expected of scheduler-card in such scenario.

I'm not expecting this to be fixed, just reporting it for completeness

> Good! This is a big improvement compared to yesterday already. Without the units you wouldn't see the '°C' I guess, only "set to 19".

Ah, you're right, I missed that. Looks better with the unit.

> You can simply assign an icon to the customized action.
> (Again, scheduler cannot guess which icon you would like to see here)

I do have the icon assigned in the customization, it is correctly displayed next to the action of the schedule configuration but still doesn't show up in the list of schedules. I had it incorrectly set to "temperature" earlier, the correct name is "thermometer":

```
      climate.bedroom_radiator:
        actions:
          - service: climate.set_temperature
            name: Set temperature
            icon: thermometer
            variables:
              temperature:
                name: Temperature
                min: 10
                max: 30
                step: 0.5
                unit: °C
```
 
![image](https://user-images.githubusercontent.com/73796046/117180702-bff35780-add4-11eb-95a0-23d62da781b6.png)


> I need a few days to shuffle things up to make things the way they were.
> Sorry for breaking things.
> I will keep you posted, hopefully you are willing to test again (and the updating goes more smoothly).

No problem at all, I have a working setup now (other than the multi-entity schedule) so that's already an improvement. Happy to test anything else you may need.

**Comment by Plawasan** [comment]
Awesome!

Customization updated:

```
    customize:
      climate.kitchen_radiator:
        actions:
          - service: climate.set_temperature
            name: Set temperature
            variables:
              temperature:
                name: Temperature
                min: 10
                max: 30
                step: 0.5
        exclude_actions:
          - heat
          - cool
```
- Icons are now set correctly both in the overview and in the action
- Units are displayed even without the definition under customize
- {action) correctly shows the target temperature

![image](https://user-images.githubusercontent.com/73796046/117352863-c4407300-aeaf-11eb-947a-6d30537c92f5.png)

Still not working (not sure if it's supposed to) - multiple entities:

![image](https://user-images.githubusercontent.com/73796046/117352929-da4e3380-aeaf-11eb-83fc-45072cd64496.png)

Btw... I know the 12/24h should be driven by the system locale but even with English-GB set I'm still seeing 12H.. I'd swear this worked correctly in an earlier version (and as far as I'm concerned the option to force this was a preferred solution..) 

For anyone who might be struggling with updates/caching what seems to do the trick for me is to delete both the .js and the .gz in the scheduler-card directory, not just the .js.. (probably a "duh!")

**Comment by nielsfaber** [comment]
> Still not working (not sure if it's supposed to) - multiple entities:

I think this is fixed now as well.

I released the changes in v2.1.2.
